first-react-app/src/components/formComponents/SubmitForm.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useEffect, useState } from "react";
+import { FieldErrors, useForm } from "react-hook-form";
+
+function SubmitForm(props: { isFormCorrect: (arg0: string[]) => void }) {
```

Comment:
you should use named interfaces

```suggestion
function SubmitForm(props: SubmitFormProps) {
```

```ts
interface SubmitFormProps {
  onSubmit: (data: SubmitData) => void;
}
```
array of strings is not OK for form
```ts
interface SubmitData {
  songName: string;
  date: string | Date;
  select: string;
  checkbox: boolean;
  radio: string;
  upload: string;
}
 ```